File: typescript-client-sdk/src/sdk/models/operations/searchcrmtasks.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class SearchCrmTasksQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=accessToken" })
  accessToken: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=allFields" })
  allFields?: boolean;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=cursor" })
  cursor?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=limit" })
  limit?: number;
}


export class SearchCrmTasksRequestBodyFilters extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=body" })
  body?: shared.StringFilter;

  @SpeakeasyMetadata({ data: "json, name=createdTime" })
  createdTime?: shared.DateFilter;

  @SpeakeasyMetadata({ data: "json, name=dueDate" })
  dueDate?: shared.DateFilter;

  @SpeakeasyMetadata({ data: "json, name=id" })
  id?: shared.StringFilter;

  @SpeakeasyMetadata({ data: "json, name=isDone" })
  isDone?: shared.BooleanFilter;

  @SpeakeasyMetadata({ data: "json, name=modifiedTime" })
  modifiedTime?: shared.DateFilter;

  @SpeakeasyMetadata({ data: "json, name=nativeId" })
  nativeId?: shared.StringFilter;

  @SpeakeasyMetadata({ data: "json, name=priority" })
  priority?: shared.StringFilter;

  @SpeakeasyMetadata({ data: "json, name=status" })
  status?: shared.StringFilter;

  @SpeakeasyMetadata({ data: "json, name=subject" })
  subject?: shared.StringFilter;
}


export class SearchCrmTasksRequestBody extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=filters" })
  filters?: SearchCrmTasksRequestBodyFilters;
}


export class SearchCrmTasksResponseBody extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=nextPageCursor" })
  nextPageCursor?: string;

  @SpeakeasyMetadata({ data: "json, name=tasks", elemType: shared.Task })
  tasks?: shared.Task[];
}


export class SearchCrmTasksRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  queryParams: SearchCrmTasksQueryParams;

  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  request?: SearchCrmTasksRequestBody;
}


export class SearchCrmTasksResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  responseBody?: SearchCrmTasksResponseBody;
}

File: go-client-sdk/pkg/models/operations/searchcrmtasks.go
package operations

import (
	"github.com/speakeasy-sdks/orb-sdks/go-client-sdk/pkg/models/shared"
)

type SearchCrmTasksQueryParams struct {
	AccessToken string   `queryParam:"style=form,explode=true,name=accessToken"`
	AllFields   *bool    `queryParam:"style=form,explode=true,name=allFields"`
	Cursor      *string  `queryParam:"style=form,explode=true,name=cursor"`
	Limit       *float64 `queryParam:"style=form,explode=true,name=limit"`
}

type SearchCrmTasksRequestBodyFilters struct {
	Body         *shared.StringFilter  `json:"body,omitempty"`
	CreatedTime  *shared.DateFilter    `json:"createdTime,omitempty"`
	DueDate      *shared.DateFilter    `json:"dueDate,omitempty"`
	ID           *shared.StringFilter  `json:"id,omitempty"`
	IsDone       *shared.BooleanFilter `json:"isDone,omitempty"`
	ModifiedTime *shared.DateFilter    `json:"modifiedTime,omitempty"`
	NativeID     *shared.StringFilter  `json:"nativeId,omitempty"`
	Priority     *shared.StringFilter  `json:"priority,omitempty"`
	Status       *shared.StringFilter  `json:"status,omitempty"`
	Subject      *shared.StringFilter  `json:"subject,omitempty"`
}

type SearchCrmTasksRequestBody struct {
	Filters *SearchCrmTasksRequestBodyFilters `json:"filters,omitempty"`
}

type SearchCrmTasksResponseBody struct {
	NextPageCursor *string       `json:"nextPageCursor,omitempty"`
	Tasks          []shared.Task `json:"tasks,omitempty"`
}

type SearchCrmTasksRequest struct {
	QueryParams SearchCrmTasksQueryParams
	Request     *SearchCrmTasksRequestBody `request:"mediaType=application/json"`
}

type SearchCrmTasksResponse struct {
	ContentType  string
	StatusCode   int64
	ResponseBody *SearchCrmTasksResponseBody
}
